client/src/components/CreatePost.tsx

```diff
@@ -0,0 +1,225 @@
+                  required
+                  fullWidth
+                  label="Body"
+                  multiline
+                  rows={6}
+                  onChange={(e) => setBody(e.target.value)}
+                />
+              </Grid>
+
+              <Grid item xs={12} md={4}>
+                <TextField
+                  required
+                  label="Upload Thumbnail"
+                  type="file"
+                  fullWidth
+                  InputLabelProps={{ shrink: true }}
+                  size="small"
+                  onChange={(e) => handleImageUpload(e)}
+                />
+              </Grid>
+
+              <Grid item xs={12} md={3}>
+                <TextField
+                  label="Event Capacity"
+                  placeholder="40"
+                  fullWidth
+                  size="small"
+                  onChange={(e) => setEventCapacity(e.target.value)}
+                />
+              </Grid>
+
+              <Grid item xs={12} md={5}>
+                <TextField
+                  fullWidth
+                  label="Location"
+                  placeholder="Zoom: https://utoronto.zoom.us/j/7227206933#success"
+                  size="small"
+                  onChange={(e) => setEventLocation(e.target.value)}
+                />
+              </Grid>
+
+              <Grid item xs={12}>
+                <TextField
+                  fullWidth
+                  label="Tags (Separate tags by a comma)"
+                  placeholder="Clubs, Math, MCS"
+                  size="small"
+                  onChange={(e) => setTags(e.target.value)}
+                />
+              </Grid>
+            </Grid>
+
+            {/* form  done */}
+            <Box sx={{ mt: 2 }}>
+              <Divider />
+            </Box>
+
+            <Grid container spacing={2} justifyContent="center">
+              <Grid item xs={5} md={3}>
+                <Button
+                  fullWidth
+                  variant="contained"
+                  sx={{ mt: 3, mb: 2 }}
+                  color="secondary"
+                  onClick={() => handleClickOpen()}
+                  size="large"
+                >
+                  Preview
+                </Button>
+              </Grid>
+              <Grid item xs={7} md={5}>
+                <Button
+                  fullWidth
+                  variant="contained"
+                  sx={{ mt: 3, mb: 2 }}
+                  size="large"
+                >
+                  Post
+                </Button>
+              </Grid>
+            </Grid>
+          </Box>
+
+          <PreviewPopUp
+            title={title}
+            body={body}
+            img={file}
+            tags={tags}
+            eventCapacity={eventCapacity}
+            location={eventLocation}
+            openPopup={openPopup}
+            handleClose={handleClose}
+          ></PreviewPopUp>
+
+          {/* Display Error */}
+
+          <Dialog open={openPopupError}>
```

Comment:
Nit - instead of an error dialog it might be better to use an alert or snackbar 

https://mui.com/components/alert/
https://mui.com/components/snackbars/